analyzer/codechecker_analyzer/buildlog/log_parser.py

```diff
@@ -848,19 +882,64 @@ def parse_options(compilation_db_entry,
     if '++' in os.path.basename(details['compiler']):
         details['lang'] = 'c++'
 
-    flag_transformers = [
+    # Source files are skipped first so they are not collected
+    # with the other compiler flags together. Source file is handled
+    # separately from the compile command json.
+    clang_flag_collectors = [
+        __skip_sources,
+        __get_output,
+        __determine_action_type,
+        __get_arch,
+        __get_language,
+        __collect_transform_include_opts,
+        __collect_clang_compile_opts
+        ]
+
+    gcc_flag_transformers = [
         __skip,
         __replace,
         __collect_compile_opts,
+        __collect_transform_include_opts,
         __determine_action_type,
         __skip_sources,
         __get_arch,
         __get_language,
         __get_output]
 
+    same_clang = False
+
+    flag_processors = gcc_flag_transformers
+    if analyzer_version_info and ('clang' in details['compiler']):
+
+        compiler_version_info = \
+            ImplicitCompilerInfo.clang_compiler_versions.get(
+                details['compiler'])
+
+        if not compiler_version_info:
+            try:
+                compiler_version_info = \
+                    get_clangsa_version_func(details['compiler'], env)
+            except subprocess.CalledProcessError as cerr:
+                LOG.error('Failed to get and parse clang version: %s',
+                          details['compiler'])
+                LOG.error(cerr)
+                compiler_version_info = None
+            ImplicitCompilerInfo.clang_compiler_versions[details['compiler']] \
+                = compiler_version_info
+
+        if compiler_version_info.installed_dir == \
+            analyzer_version_info.installed_dir or \
+            compiler_version_info.major_version == \
```

Comment:
maybe this one is too strict too. We cannot handle it any better if the the major versions of the compiler and the analyzer are different. 

So I suggest to to just exampine only if original compiler is clang.: The first word is clang of the ouptut of clang --version.

ednikru@kojak:/mnt/ssd/ednikru/codechecker$ clang --version
clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
Target: x86_64-pc-linux-gnu